### PR TITLE
raftstore-v2: implement force leader

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Support force for raftstore v2.

Issue Number: ref #15108

What's Changed:

```commit-message
* Add UnsafeRecoveryHandle trait to abstract unsafe recovery APIs
* Implement UnsafeRecoveryHandle for raftstore v1 router
* Implement UnsafeRecoveryHandle for raftstore v2 router
* Add unsafe recovery messages to v2 Store and v2 Peer
* raftstore-v2: implement force leader 
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]

File: tests/integrations/raftstore/test_unsafe_recovery.rs [hunk 1]
```diff
@@ -1215,9 +1231,10 @@ fn test_unsafe_recovery_has_commit_merge() {
     assert!(has_commit_merge);
 }
 
-#[test]
+#[test_case(test_raftstore::new_node_cluster)]
+// #[test_case(test_raftstore_v2::new_node_cluster)]
```
Comment (Connor1996) [inline tests/integrations/raftstore/test_unsafe_recovery.rs:1235]: why comment out

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

File: tests/integrations/raftstore/test_unsafe_recovery.rs (hunk 1 above)
Comment (overvenus) [reply tests/integrations/raftstore/test_unsafe_recovery.rs:1235]: Will enable it later, for now demote is not supported yet.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**